### ISSUE TITLE
[Subtask] save the SQL session metadata into MetadataManager

### ISSUE THREAD

**Issue by lightning-L** [apache/incubator-kyuubi#3723]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

save the SQL session metadata into MetadataManager

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/3554

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to improve.
- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

**Comment by lightning-L** [comment]
1. save the SQL session metadata into MetadataManager
2. list the live sessions from the MetadataManager
3. support to list sessions with filter condition, such as user

**Comment by EricGao888** [comment]
@lightning-L Hi, this feature seems significant when using kyuubi restful sessions in HA deployment. May I ask why it is not merged into main branch? Thanks : )

**Comment by z1131392774** [comment]
Hi @pan3793 @yaooqinn ,sorry to bother you.

Following up on my previous comment, I'd like to take over this inactive issue. I have some initial thoughts on the implementation and would appreciate your feedback.

Here are my preliminary ideas:

1. SessionState Definition

I propose creating a new SessionState enum, similar to OperationState. My initial thought is to include the following states:

- INITIALIZED: Session is initialized.
- RUNNING: Session is active and running operations.
- PENDING: Session is pending, waiting for the engine to start.
- CLOSED: Session is closed by a user request.
- TIMEOUT: Session is closed due to inactivity timeout.
- ERROR: Session terminated with an error.

Are these states sufficient, or do you have other suggestions?

2. Fields to Persist for Session Metadata

To store session metadata in the MetadataStore, I think we should persist the following core fields, which are essential for management and troubleshooting:

- Session Identity: identifier, sessionType
- User & Connection: realUser, username, ipAddress
- State & Timestamps: state, createTime, endTime
- Kyuubi Instance: kyuubiInstance
- Configuration: requestConf
- Engine Details: engineType, engineId, engineName, engineUrl, engineError
- Session/Job Name: requestName

Does this list seem complete? Is there anything to add or remove?

I will wait for your feedback before starting the work.

By the way, should I open a new issue for this implementation, or is it fine to continue the discussion here? Thanks!

**Comment by yaooqinn** [comment]
cc @turboFei
